### PR TITLE
Fix/challenge log

### PR DESCRIPTION
**Pull request by cnlangzi**
- improved logging on blobber/challenge
- fixed VerifyMerklePath bug on validator

need merge https://github.com/0chain/gosdk/pull/269

### CODE REVIEW

File: code/go/0chain.net/blobbercore/allocation/newfilechange.go [hunk 1]
```diff
@@ -22,9 +22,9 @@ type NewFileChange struct {
 	//client side:
 	Path string `json:"filepath" validation:"required"`
 	//client side:
-	ActualHash string `json:"actual_hash,omitempty" validation:"required"`
```
Comment (kushthedude) [inline code/go/0chain.net/blobbercore/allocation/newfilechange.go:25]: Are these fields not mandatory 🤔

**Review by kushthedude — DISMISSED** [review]
Looks good overall, just one query

File: code/go/0chain.net/blobbercore/allocation/newfilechange.go (hunk 1 above)
Comment (cnlangzi) [reply code/go/0chain.net/blobbercore/allocation/newfilechange.go:25]: yes. We don't know it if input is a stream.

**Comment by cnlangzi** [comment]
only fixed go.mod for staging.

**Pull request merged**